consensus/serde_utils/src/quoted_u64_var_list.rs

```diff
@@ -0,0 +1,128 @@
+//! Formats `VariableList<u64,N>` using quotes.
+//!
+//! E.g., `VariableList::from(vec![0, 1, 2])` serializes as `["0", "1", "2"]`.
+//!
+//! Quotes can be optional during decoding. If `N` is greater than the length of the `Vec`, the `Vec` is truncated.
+
+use serde::ser::SerializeSeq;
+use serde::{Deserializer, Serializer};
+use serde_derive::{Deserialize, Serialize};
+use ssz_types::typenum::Unsigned;
+use ssz_types::VariableList;
+use std::marker::PhantomData;
+
+#[derive(Serialize, Deserialize)]
+#[serde(transparent)]
+pub struct QuotedIntWrapper {
+    #[serde(with = "crate::quoted_u64")]
+    pub int: u64,
+}
+
+pub struct QuotedIntVarListVisitor<N> {
+    _phantom: PhantomData<N>,
+}
+
+impl<'a, N> serde::de::Visitor<'a> for QuotedIntVarListVisitor<N>
+where
+    N: Unsigned,
+{
+    type Value = VariableList<u64, N>;
+
+    fn expecting(&self, formatter: &mut std::fmt::Formatter) -> std::fmt::Result {
+        write!(formatter, "a list of quoted or unquoted integers")
+    }
+
+    fn visit_seq<A>(self, mut seq: A) -> Result<Self::Value, A::Error>
+    where
+        A: serde::de::SeqAccess<'a>,
+    {
+        let mut vec = vec![];
+
+        while let Some(val) = seq.next_element()? {
+            let val: QuotedIntWrapper = val;
+            vec.push(val.int);
+        }
+        let fix: VariableList<u64, N> = VariableList::from(vec);
```

Comment:
As per previous comment, I think it would be safest to check we don't exceed the `N` length.